### ISSUE TITLE
ADA | Axe Issue: Pop-up Menu Not Contained by Landmarks

### ISSUE THREAD

**Issue by kalyanighodake** [jpmorganchase/salt-ds#4504]
### Latest version

- [X] I have tested the latest version

### Description

When opening the menu pop-up using the Salt component, the Axe accessibility tool reports an error: "All page content should be contained by landmarks." This issue occurs because the menu dropdown is rendered in a data-floating-ui-portal outside the <main> element, which may cause it to be perceived as not being part of the structured content of the page.

### Steps to reproduce

1. Navigate to the page containing the Salt component with the menu pop-up.
2. Trigger the menu pop-up to open.
3. Run the Axe accessibility tool to analyze the page.
4. Observe the reported issue regarding content not being contained by landmarks.

### Expected behavior

All page content, including dynamically rendered elements like pop-up menus, should be logically associated with a landmark region to ensure accessibility compliance.

### Package name(s)

Core (@salt-ds/core)

### Package version(s)

"@salt-ds/core": "^1.37.0"

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
At first I wasn't able to reproduce this, I had to enable "Best practices" when scanning

**Comment by jake-costa** [comment]
@kalyanighodake 
Thank you for highlighting this issue! The challenge with the Axe accessibility tool's report is indeed related to our use of portaling, like Josh mentioned. We use this technique to ensure overlays, like our menu pop-ups, are always displayed correctly above other content.

In this specific case, the use of portaling is considered an accessibility exception, similar to how modal dialogs and some ARIA live regions are handled. These elements are intentionally placed outside their invoking elements in the DOM to function properly. Our commitment to accessibility involves using a variety of best practices methods and thoroughly testing our solutions across multiple assistive technology and browser setups. However, if you experience any issues using assistive technology, please don't hesitate to let us know.

I see that you are an employee of JPMorgan Chase & Co., so I recommend taking a look at our DAKB 1.3.1-09 for more detailed information on how we handle these scenarios.

That said, we will still review this setup to see if there are further improvements we can make to enhance accessibility without compromising functionality. 

Thanks again for your valuable feedback. We appreciate your help in making our components better!
